### ISSUE TITLE
ImageCache.storeToDisk() throws a NullPointerException unpredicatedly.

### ISSUE THREAD

**Issue by danielsunyu** [applidium/Shutterbug#6]
Hello there,

I've integrated the Shutterbug into a commercial app of my company to tackles the economic image-loading & caching issue for several months. Initially through the exception log of flurry, I found out that storeToDisk() crashes occasionally. After putting a general exception-catch block, the bug is fixed. And I want to share it with all.

Note that I put comment near the code I inserted. Hope it will help others. And dear author of the library, please consider this.

```
public Snapshot storeToDisk(InputStream inputStream, String cacheKey) {
    try {
        Editor editor = mDiskCache.edit(cacheKey);
        final OutputStream outputStream = editor.newOutputStream(0);
        final int bufferSize = 1024;
        try {
            byte[] bytes = new byte[bufferSize];
            for (;;) {
                int count = inputStream.read(bytes, 0, bufferSize);
                if (count == -1)
                    break;
                outputStream.write(bytes, 0, count);
            }
            outputStream.close();
            editor.commit();
            return mDiskCache.get(cacheKey);
        } catch (Exception e) {
            e.printStackTrace();
        }
    } catch (IOException e) {
        e.printStackTrace();
    } catch (Exception e) { // Danny added a catch subclause here to prevent NullPointerException.
        e.printStackTrace();
    }
    return null;
}
```

**Comment by johnxe** [comment]
The variable editor may be null.

**Comment by tokou** [comment]
Hi,
This has been solved in commit 9766d3b41d23f91bc62cb34db0c6c6f66fc0d01b
